[DOC] Add `AptosToken` and `FungibleAssetClient` plugin classes to the doc site. (#8779)
aptos-labs/aptos-core: developer-docs-site/docs/sdks/ts-sdk/sdk-plugins-layer.md

A plugin is a component that can be added to the TypeScript SDK to extend or enhance its functionality. Plugins are meant to be built to support popular applications on the Aptos network and can be used to add new features, ease the use of the application operations and to customize the user experience.

## AptosToken class

> The [AptosToken](https://github.com/aptos-labs/aptos-core/blob/main/ecosystem/typescript/sdk/src/plugins/aptos_token.ts) class compatible with the [token V2 standard](../../standards/aptos-token-v2.md) and provides methods for creating and querying NFT collections and tokens.

Reviewer: ```suggestion
The [AptosToken](https://github.com/aptos-labs/aptos-core/blob/main/ecosystem/typescript/sdk/src/plugins/aptos_token.ts) class is compatible with the [token V2 standard](../../standards/aptos-token-v2.md) and provides methods for creating and querying NFT collections and tokens.
```

It covers write methods that support creating, transferring, mutating, and burning tokens on-chain

The main write methods supported by the AptosToken class are:

- Create Collection
- Mint
- Mint Soul Bound
- Burn Token
- Freeze Token Transafer

> - Unfreeze Token Transafer

Reviewer 2: Transfer

- Get TokenData
- Get Token of an Account

## FungibleAssetsClient class

The [FungibleAssetsClient](https://github.com/aptos-labs/aptos-core/blob/main/ecosystem/typescript/sdk/src/plugins/fungible_asset_client.ts) class compatible with the [fungible asset component](../../standards/fungible-asset.md) and provides methods to transfer fungible assets between accounts and to check an account balance.

The main write methods are:

- Transfer

> - Generate Transfer

Reviewer 2: What's generate transfer?

Author: generate transfer transaction https://github.com/aptos-labs/aptos-core/blob/main/ecosystem/typescript/sdk/src/plugins/fungible_asset_client.ts#L71

The main read methods are:

- Get Primary Balance

## CoinClient class

> The [CoinClient](https://github.com/aptos-labs/aptos-core/blob/main/ecosystem/typescript/sdk/src/plugins/coin_client.ts) class provides methods to intercat with the coin module to transfer coins between accounts and to check an account balance. By default it transfers `0x1::aptos_coin::AptosCoin`, but you can specify a different coin type with the `coinType` argument.

Reviewer: ```suggestion
The [CoinClient](https://github.com/aptos-labs/aptos-core/blob/main/ecosystem/typescript/sdk/src/plugins/coin_client.ts) class provides methods to interact with the coin module to transfer coins between accounts and to check an account balance. By default, it transfers `0x1::aptos_coin::AptosCoin`, but you can specify a different coin type with the `coinType` argument.
```

- Fund Account

## ANSClient class

The [ANSClient](https://github.com/aptos-labs/aptos-core/blob/main/ecosystem/typescript/sdk/src/plugins/ans_client.ts) class provides methods for creating an ANS name on the Aptos network and querying ANS data.

> It covers (1) write methods that support creating a unique identity on the Aptos network by registering an Aptos name and (2) read methods to get an account ANS name by an account address and get an account address by an ANS name.

Reviewer: ```suggestion
It covers (1) write methods that support creating a unique identity on the Aptos network by registering an Aptos name and (2) read methods that retrieve an account's ANS name using its address, as well as retrieving an account's address using its ANS name.
```
